### ISSUE TITLE
Closing certain pages (drawers) navigates to the wrong URL

### ISSUE THREAD

**Issue by jasperhuangg** [Expensify/App#11111]
#### Connect bank account page (https://github.com/Expensify/App/issues/10939):
1. Open web app https://new.expensify.com/
2. Click on user icon on LHN to open setting.
3. Click on user's workspace.
4. Click on Connect bank account.
5. Click on close icon on top right.

#### Default room participant detail page (https://github.com/Expensify/App/issues/10989):
1. Go to staging.new.expensify.com and log in with any account
2. Click on any default room (workspace room)
3. Click on header to open details page > settings > go back by clicking back button >
4. Click on members > any member.
5. Click on close button

#### Paypal.me payments page (https://github.com/Expensify/App/issues/10991):
1. Go to staging.new.expensify.com
2. Log in with any account
3. Open settings page > Workspace > Manage members
4. Select Invite member and close the page
5. Open settings page > payments > Add payment method > Paypal
6. Close the page

#### Closing App download links page in settings page is causing incorrect URL settings/about instead of r/reportId https://github.com/Expensify/App/issues/11350#
1. Login in web app.
2. Open settings page > profile > click add phone number
3. Click on back arrow on this page to navigate to profile page.
4. Click on back arrow to navigate to settings page.
5. Click on about > app download links.
6. Click on cross (x) to close app download links page.

## Expected Result:
The URL should display correctly

## Actual Result:
The URL is incorrect. Namely, the URL that is shown should have already been popped off of the navigation stack (is it the URL for a previous page).

## Workaround:
Yes

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:** 
**Issue reported by:** @aneequeahmad (from https://github.com/Expensify/App/issues/10989)
**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

cc @parasharrajat

**Comment by aneequeahmad** [comment]
I see all of these issues are combined so it's easy to track them for `engineers and QA team`. Just a `quick question` would it will be considered `one bug bounty` for reporting all of these issues ? Almost all the bugs are reported by me and I have spent a lot of time on debugging and finding these issues. 

cc: @parasharrajat, @flodnv

**Comment by flodnv** [comment]
@jasperhuangg any thoughts? And why is this on HOLD? 😕

**Comment by jasperhuangg** [comment]
@flodnv @aneequeahmad apologies for the lack of response, was out sick last week 

I think we can still compensate you for reporting each of the issues separately, since I do recognize that it requires some time to find them even if they have the same source.

@parasharrajat recommended we HOLD this issue on navigation, can you clarify why?

**Comment by parasharrajat** [comment]
@jasperhuangg New design doc for navigation is in planning that's why I suggested holding this like other navigation issues. 

it seems like @puneetlath  kick-started it https://expensify.slack.com/archives/C01GTK53T8Q/p1664825958481829.

**Comment by MitchExpensify** [comment]
I'm a little confused on what to do with this collection of issues. Mind pointing me in the right direction @mallenexpensify ?

**Comment by mallenexpensify** [comment]
There is no direction Mitch!!!!
@jasperhuangg is this more of a tracking issue right now? I'm unsure if anything is currently actionable

**Comment by michaelhaxhiu** [comment]
IMO this shouldn't be `daily` if it's on hold. It should go to `monthly` till it's ready to move on it

**Comment by JmillsExpensify** [comment]
We discussed in Slack, but to reiterate here, @puneetlath is trying new methods to get in contact with the maintainer. We're unfortunately dead in the water and still on hold until we can get in touch.

**Comment by JmillsExpensify** [comment]
Also just realized that this issue isn't linked to the tracking issue. Adding that here as well as the title. https://github.com/Expensify/App/issues/11768

**Comment by JmillsExpensify** [comment]
Still blocked on clear next steps in the main tracking issue.

**Comment by JmillsExpensify** [comment]
Still on hold.

**Comment by JmillsExpensify** [comment]
Yes, but this is on hold Melvin. Switching out daily for weekly.

**Comment by JmillsExpensify** [comment]
Still holding.

**Comment by JmillsExpensify** [comment]
Still held as we go through the design process.

**Comment by JmillsExpensify** [comment]
Still held on the larger react navigation reboot.

**Comment by JmillsExpensify** [comment]
Still on hold for the larger navigation reboot.

**Comment by JmillsExpensify** [comment]
Still on hold for navigation.

**Comment by JmillsExpensify** [comment]
Held on navigation revamp.

**Comment by JmillsExpensify** [comment]
Still held on the navigation revamp.

**Comment by JmillsExpensify** [comment]
Still on hold for navigation. No ETA.

**Comment by JmillsExpensify** [comment]
Same same.

**Comment by JmillsExpensify** [comment]
Still on hold.

**Comment by JmillsExpensify** [comment]
Still on hold

**Comment by JmillsExpensify** [comment]
Coming off hold very soon!

**Comment by JmillsExpensify** [comment]
Alright, I've gone through each of the linked issues, run through the reproduction steps, confirmed that they are all resolved. I'm handling next steps in the linked issues, closing this main tracking issue out as a result.
